astroplan/core.py

```diff
@@ -0,0 +1,490 @@
+        """
+        Returns the local moonset time.
+
+        The default moonset returned is the next one to occur.
+
+        Parameters
+        ----------
+        date_time : WHAT TYPE IS date_time OBJECT?
+
+        Keywords: str, optional
+            previous
+            next
+        """
+        raise NotImplementedError()
+
+    def moon_illumination(date_time):
+        """
+        Returns a float giving the percent illumation.
+
+        Parameters
+        ----------
+        date_time : WHAT TYPE IS date_time OBJECT?
+        """
+        raise NotImplementedError()
+
+    def moon_position(date_time):
+        """
+        Returns the position of the moon in alt/az.
+
+        Parameters
+        ----------
+        date_time : WHAT TYPE IS date_time OBJECT?
+        """
+        raise NotImplementedError()
+
+    # Other time-related methods.
+
+    def evening_nautical(date_time):
+        """
+        Returns the local evening (nautical) twilight time.
+
+        Parameters
+        ----------
+        date_time : WHAT TYPE IS date_time OBJECT?
+        """
+        raise NotImplementedError()
+
+    def evening_civil(date_time):
+        """
+        Returns the local evening (civil) twilight time.
+
+        Parameters
+        ----------
+        date_time : WHAT TYPE IS date_time OBJECT?
+        """
+        raise NotImplementedError()
+
+    def morning_astronomical(date_time):
+        """
+        Returns the local morning (astronomical) twilight time.
+
+        Parameters
+        ----------
+        date_time : WHAT TYPE IS date_time OBJECT?
+        """
+        raise NotImplementedError()
+
+    def morning_nautical(date_time):
+        """
+        Returns the local morning (nautical) twilight time.
+
+        Parameters
+        ----------
+        date_time : WHAT TYPE IS date_time OBJECT?
+        """
+        raise NotImplementedError()
+
+    def morning_civil(date_time):
+        """
+        Returns the local morning (civil) twilight time.
+
+        Parameters
+        ----------
+        date_time : WHAT TYPE IS date_time OBJECT?
+        """
+        raise NotImplementedError()
+
+
+class Target(object):
+    """
+    This is an abstract base class -- you can't instantiate
+    examples of this class, but must work with one of its
+    subclasses such as `FixedTarget` or `NonFixedTarget`.
+
+    Would need to import six, abc to make this a metaclass?
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, name, ra, dec, marker=None):
+        """
+        Defines a single observation target.
+
+        Parameters
+        ----------
+        name : str, optional
+
+        ra : WHAT TYPE IS ra ?
+
+        dec : WHAT TYPE IS dec ?
+
+        marker : str, optional
+            User-defined markers to differentiate between different types
+            of targets (e.g., guides, high-priority, etc.).
+        """
+        raise NotImplementedError()
+
+    @property
+    def ra(self):
```

Comment:
These should be removed in favor of a `.coord` attribute (which doesn't need to be a property defined like this -- as simple attribute is probably ok, unless we want to make it read-only).